### PR TITLE
refcount: Use `performBlockingMajorGC` if available

### PR DESCRIPTION
**Pull request by wenkokke**

### CODE REVIEW

**Comment by wenkokke** [comment]
@dcoutts This uses `performBlockingMajorGC` when available and uses your trick otherwise. (I've removed the first assert, since asserting twice seems superfluous.)

**Comment by wenkokke** [comment]
@dcoutts This still requires `yield` in order to pass the test suite, even when using `performBlockingMajorGC`.

**Comment by wenkokke** [comment]
From cursory reading of GHC source, ` performBlockingMajorGC` appears mostly to be the correct alternative to use for compatibility with non-moving concurrent GC. So we should probably keep your hack and combine it with preferring `performBlockingMajorGC` if it’s available?

**Review by dcoutts — COMMENTED** [review]
I think this whole thing could be a lot simpler.

How about:
```
import qualified System.Mem
```
and then
```
-- Later GHC versions have 'performBlockingMajorGC' which does a full GC even
-- when using the concurrent incremental collector. This is what we want to ensure
-- all forgotten refs are found.
#if MIN_VERSION_base(4,20,0)
performBlockingMajorGC = System.Mem.performBlockingMajorGC
#else
performBlockingMajorGC = System.Mem.performMajorGC
#endif
```
and otherwise in `checkForgottenRefs`, keep it the same as originally but replace `performMajorGC` with `performBlockingMajorGC`. And the comment you added is good too.

But there's nothing magic about `performBlockingMajorGC` that means we don't need yield or that means we should do it once vs twice. The `performBlockingMajorGC` is just the same as `performMajorGC` in older GHC versions except that it provides the same guarantee for the incremental GC as the non-incremental: that it scans the whole heap.

**Review by dcoutts — APPROVED** [review]
LGTM now. Please squash and merge.

File: src-control/Control/RefCount.hs [hunk 1]
```diff
@@ -468,12 +472,24 @@ checkForgottenRefs = do
 #ifndef NO_IGNORE_ASSERTS
     return ()
 #else
-    performMajorGC
+    -- The hope is that by combining `performMajorGC` with `yield` that
+    -- the former starts the GC threads and the latter puts the current
```
Comment (dcoutts) [inline src-control/Control/RefCount.hs:476]: I suppose technically the former starts weak pointer finaliser threads, not GC threads.

**Review by dcoutts — APPROVED** [review]

**Comment by dcoutts** [comment]
Unclear what's up with stylish-haskell.

File: src-control/Control/RefCount.hs (hunk 1 above)
Comment (wenkokke) [reply src-control/Control/RefCount.hs]: Fixed and clarified the comment in general.

**Pull request merged**